code/game/objects/structures/crates_lockers/closets/secure/security.dm

```diff
@@ -255,100 +255,114 @@
 	icon_broken = "seccadetbroken"
 	icon_off = "seccadetoff"
 
-	fill()
-		//Appearance
-		if(prob(50))
-			new /obj/item/weapon/storage/backpack/security(src)
-		else
-			new /obj/item/weapon/storage/backpack/satchel_sec(src)
-		new /obj/item/clothing/head/beret/sec/cadet(src)
-		new /obj/item/clothing/suit/storage/vest/cadet(src)
-		new /obj/item/clothing/under/rank/cadet(src)
-		//Tools
-		new /obj/item/device/radio/headset/headset_sec(src)
-		new /obj/item/device/flash(src)
-		new /obj/item/weapon/reagent_containers/spray/pepper(src)
-		new /obj/item/taperoll/police(src)
-		new /obj/item/device/hailer(src)
-		new /obj/item/device/holowarrant(src)
-		new /obj/item/device/flashlight/flare(src)
-		//Belts
-		if (prob(50))
-			new /obj/item/clothing/accessory/storage/black_vest(src)
-		else
-			new /obj/item/clothing/accessory/storage/pouches/black(src)
-		new /obj/item/weapon/storage/belt/security(src)
+/obj/structure/closet/secure_closet/security_cadet/fill()
+	//Appearance
+	if(prob(50))
+		new /obj/item/weapon/storage/backpack/security(src)
+	else
+		new /obj/item/weapon/storage/backpack/satchel_sec(src)
+	new /obj/item/clothing/head/beret/sec/cadet(src)
+	new /obj/item/clothing/suit/storage/vest/cadet(src)
+	new /obj/item/clothing/under/rank/cadet(src)
+	//Tools
+	new /obj/item/device/radio/headset/headset_sec(src)
+	new /obj/item/device/flash(src)
+	new /obj/item/weapon/reagent_containers/spray/pepper(src)
+	new /obj/item/taperoll/police(src)
+	new /obj/item/device/hailer(src)
+	new /obj/item/device/holowarrant(src)
+	new /obj/item/device/flashlight/flare(src)
+	//Belts
+	if (prob(50))
+		new /obj/item/clothing/accessory/storage/black_vest(src)
+	else
+		new /obj/item/clothing/accessory/storage/pouches/black(src)
+	new /obj/item/weapon/storage/belt/security(src)
 
 /obj/structure/closet/secure_closet/security
 	name = "security officer's locker"
-	req_access = list(access_brig)
+	req_access = list(access_weapons)

```

Comment:
Dont use `access_weapons` for that.
`access_weapons` is used to determine if someone is allowed to use weapons, not if they are a security officer.